kernel/src/table_changes/log_replay.rs

```diff
@@ -181,13 +181,10 @@ impl LogReplayScanner {
             }
             if let Some((schema, configuration)) = visitor.metadata_info {
                 let schema: StructType = serde_json::from_str(&schema)?;
-                // Currently, schema compatibility is defined as having equal schema types. In the
-                // future, more permisive schema evolution will be supported.
-                // See: https://github.com/delta-io/delta-kernel-rs/issues/523
-                require!(
-                    table_schema.as_ref() == &schema,
-                    Error::change_data_feed_incompatible_schema(table_schema, &schema)
-                );
+
+                schema.can_read_as(table_schema).map_err(|err| {
+                    Error::change_data_feed_incompatible_schema(commit_file.version, err)
+                })?;
```

Comment:
right so the two places that are changing is: (1) start time we check start/end schema compat and (2) as we process metadata changes (here) we check for schema compat?

Reply:
Yep, correct